### PR TITLE
[docs] Trash

### PR DESCRIPTION
**Pull request by alexyarosh**
Rename archive to Trash, pull it out of History, and add explanations of how trashing affects related items.

Not sure if it should have a backport label considering RC1 was cut

Things to think about:

Name and content:
- This goes into “Organization” topic so I thought “”the means/goal of organization action” (delete and restore) would be a better name than ”the place where it goes” (Trash).
But then this also made me think that people might come to that article to learn about deleting items that don’t go into trash, so I added a bunch of links to other delete-like actions.

Article organization:

- There are a lot of repeated instructions which I don’t like. On the other hand, I wanted to have detailed section titles for better searcheability. Alternative would be to have delete/restore instructions as one section, and then separate the effects of delete actions into sections.

Downstream effects:
- I think I noted most of the possible consequences/interactions of deleting something, but let me know if I missed something!

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -126,6 +126,7 @@ Metabase's reference documentation.
 - [Basic exploration](./exploration-and-organization/exploration.md)
 - [Collections](./exploration-and-organization/collections.md)
 - [History](./exploration-and-organization/history.md)
+- [Delete and restore](./exploration-and-organization/delete-and-restore.md)
```
Comment (jeff-bruemmer) [inline docs/README.md:129]:
```suggestion
- [Trash](./exploration-and-organization/delete-and-restore.md)
```

I get that you're going for keywords `delete` and `restore`, but we can use trash here, as that's what the page covers.

File: docs/exploration-and-organization/delete-and-restore.md [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+title: Delete and recover items
+---
+
+# Delete and recover items
+
+Sometimes your questions, dashboards, or models outlive their usefulness. You can send outdated items to **Trash**.
+
+![Move to trash](./images/move-to-trash.png)
+
+Items in **Trash** won't show up in search, and you won't be able to use them to create new questions and dashboards.
```
Comment (jeff-bruemmer) [inline docs/exploration-and-organization/delete-and-restore.md:11]: You'll need to note here that you can search for items in the trash, if you need to.

File: docs/exploration-and-organization/delete-and-restore.md [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+3. Click on the three dot menu;
+4. Select "Restore".
+
+Deleting a model will affect other items that depend on that model:
+
+- **Derived questions**: If a trashed model was used as data source for a Query Builder question, or was [referenced in a SQL question](../questions/native-editor/referencing-saved-questions-in-queries.md), then the derived questions will continue to work while the model is in Trash. If the model is [permanently deleted](#permanently-deleting-items), the derived questions will stop working, and you'll get a `Card does not exist` error.
+
+- **Actions**: If a model is trashed, the actions in this model will still work. If a model is permanently deleted, the actions in the deleted model will be deleted as well.
+
+- **Dashboards**: If a model is included as a card in any dashboards, then the card will be removed from those dashboards when the model is moved to Trash. If the model is restored from Trash, its card will be restored to the dashboards that used to contain it.
```
Comment (jeff-bruemmer) [inline docs/exploration-and-organization/delete-and-restore.md:115]: The text is confusing. maybe a table? Columns are: Model in collection, Trash, permanently deleted.

So something like: 

**Item** |**In trash**|**Permanently deleted**|**Restored**
-----|-----|-----|-----
Question based on model|Works as normal|Breaks|Works as normal
Dashboard that includes model|Model card is removed from the dashboard| Model card is removed from the dashboard |Card is restored to the dashboard

**Review by jeff-bruemmer — CHANGES_REQUESTED** [review]
- We should prioritize human eyeballs first, then machine eyeballs. To that end, we could significantly consolidate these docs. I'm not categorically against duplication, but here the repetition makes it easy to see the trees, but not the forest.
- The forest: we should talk about how to delete and restore items in general, then list the effects of a trash move for each type of item. I doubt many people will ever search for how to delete something. What people might want to read is info about what happens to related items when they delete something. E.g., "what happens to items and sub-collections when you move a collection to the trash", or dashboard cards/subscriptions, or models with downstream questions, etc.
- In general, use existing docs and expand them, rather than repeat elsewhere. E.g., the search section should be a one-liner that links to the [search section](https://github.com/metabase/metabase/blob/master/docs/exploration-and-organization/exploration.md#advanced-search) (this linking will eliminate at least one screenshot).
- Speaking of screenshots: I prefer a little more visual context in the images. Right now the menus are isolated/disconnected from the larger UI so they're difficult to situate in context. And since the menus are straightforward, it's more important to show people where to find the menus, not how to use them.
- I know our table support isn't stellar, but there are some opportunities for them here to enumerate item cases (I gave one hasty example in-line re: model trashing).

I know I'm asking for a lot of changes, but this is an excellent first draft, @alexyarosh. Very thorough. Made it easy to edit and identify the important stuff we should highlight.

**Comment by alexyarosh** [comment]
Thanks for awesome feedback @jeff-bruemmer ! Check out the new draft

File: docs/exploration-and-organization/delete-and-restore.md [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+
+You;'ll be able to see the contents of deleted dashboards, questions, and models in Trash, but you won't be able to modify them.
+
+## Search in Trash
+
+You can use [advanced search](./exploration.md) with a "Search items in trash" toggle to find items in Trash
+
+## Deleting and restoring items
+
+To move an item (question, dashboard, model, or collection) to Trash:
```
Comment (jeff-bruemmer) [inline docs/exploration-and-organization/delete-and-restore.md:33]:
```suggestion
To move an item (question, dashboard, model, or collection) to the Trash:
```

File: docs/exploration-and-organization/delete-and-restore.md [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+> Restoring a collection will also restore all the items from that collection.
+
+If the item's original parent collection has been deleted as well, you won't see an option to **Restore**. You'll still be able to move the it from Trash to a different collection.
+
+## How deleting an item affects related items
+
+Deleting or restoring an item will affect other items that depend on that item.
+
+### Questions
+
```
Comment (jeff-bruemmer) [inline docs/exploration-and-organization/delete-and-restore.md:59]: Maybe just a intro sentence for each table for more context? Something like:  "What happens to related items when you delete a question?"

File: docs/exploration-and-organization/delete-and-restore.md [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+## Search in Trash
+
+You can use [advanced search](./exploration.md) with a "Search items in trash" toggle to find items in Trash
+
+## Deleting and restoring items
+
+To move an item (question, dashboard, model, or collection) to Trash:
+
+1. Go to the question you want to delete;
```
Comment (jeff-bruemmer) [inline docs/exploration-and-organization/delete-and-restore.md:35]: Think we need to to also talk about trashing items from the collection view, in particular bulk trashing.

File: docs/exploration-and-organization/delete-and-restore.md [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+
+## Collections
+
+| Affected item                                  | In Trash | Permanently deleted | Restored |
+| ---------------------------------------------- | -------- | ------------------- | -------- |
+| All items and subcollecitons in the collection | In Trash | Permanently deleted | Restored |
+
+## Permanently deleting items
+
+Moving an item to Trash is not permanent: you'll be able to restore it to the original parent collection, or move it to a non-deleted collection.
```
Comment (jeff-bruemmer) [inline docs/exploration-and-organization/delete-and-restore.md:89]:
```suggestion
Moving an item to Trash doesn't delete the item completely: you'll be able to restore the item from the Trash.
```
Slight clarification.

File: docs/exploration-and-organization/delete-and-restore.md [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+
+To permanently delete an item:
+
+1. Go to Trash;
+2. Find the item you'd like to delete;
+3. Click on the checkbox next to the collection to select it;
+4. Select "Permanently delete".
```
Comment (jeff-bruemmer) [inline docs/exploration-and-organization/delete-and-restore.md:96]: Maybe some warning about how you won't be able to recover the item. It's lost to the void.

File: docs/exploration-and-organization/delete-and-restore.md [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+
+Events and timelines can be archived and unarchived. See [Archiving Events and timelines](events-and-timelines.md#archiving-timelines).
+
+You won't see archived Events and Timelines in Trash. To see archived events and timelines, you need to [access them from the collection's page](events-and-timelines.md#view-archived-events-and-timelines).
+
+## Deleting and restoring SQL snippets
+
+SQL snippets can be archived and unarchived. See [Archiving SQL snippets](../questions/native-editor/sql-snippets.md#archiving-snippets).
+
+You won't see archived SQL snippets in Trash. To see archived SQL Snippets, you need to [access them from the cSnippet menu](../questions/native-editor/sql-snippets.md#snippet-menu).
```
Comment (jeff-bruemmer) [inline docs/exploration-and-organization/delete-and-restore.md:108]:
```suggestion
You won't see archived SQL snippets in Trash. To see archived SQL Snippets, you need to [access them from the Snippet menu](../questions/native-editor/sql-snippets.md#snippet-menu).
```

**Review by jeff-bruemmer — APPROVED** [review]
Left a few comments to address before merging.

And remember to remove unused images from the PR.

Nice work.

**Comment by alexyarosh** [comment]
Trash is not making it to v50 so putting this on hold for now

File: docs/exploration-and-organization/delete-and-restore.md [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+
+## Deleting and restoring SQL snippets
+
+SQL snippets can be archived and unarchived. See [Archiving SQL snippets](../questions/native-editor/sql-snippets.md#archiving-snippets).
+
+You won't see archived SQL snippets in Trash. To see archived SQL Snippets, you need to [access them from the Snippet menu](../questions/native-editor/sql-snippets.md#snippet-menu).
+
+## Deleting metrics and segments
+
+Metrics and events can be retired. See [Retiring Metrics and Segments](../data-modeling/segments-and-metrics.md#editing-and-retiring-segments-and-metrics).
```
Comment (appleby) [inline docs/exploration-and-organization/delete-and-restore.md:126]: Hi @alexyarosh, this link is failing the `yarn run docs-lint-links` stage for me in CI.

Unclear to me what the link target should be. Did you forget to `git add` the `segments-and-metrics.md` file, or should the link maybe point to `segments.md` instead?

Comment (appleby) [reply docs/exploration-and-organization/delete-and-restore.md:126]: Link to the CI failure for reference

https://github.com/metabase/metabase/actions/runs/11227506788/job/31245666789

Comment (alexyarosh) [reply docs/exploration-and-organization/delete-and-restore.md:126]: @appleby done #48478

Comment (appleby) [reply docs/exploration-and-organization/delete-and-restore.md:126]: Ah makes, sense. Thanks!

**Pull request merged**